tensorflow/c/experimental/filesystem/plugins/hadoop/hadoop_filesystem.cc

```diff
@@ -209,14 +209,15 @@ hdfsFS Connect(tf_hadoop_filesystem::HadoopFile* hadoop_file,
   }
   if (hadoop_file->connection_cache.find(cacheKey) ==
       hadoop_file->connection_cache.end()) {
-    hadoop_file->connection_cache[cacheKey] =
-        libhdfs->hdfsBuilderConnect(builder);
+    auto cacheFs = libhdfs->hdfsBuilderConnect(builder);
```

Comment:
We need to take a lock here as well, right?

Reply:
I will take care of the filesystem plugin later